tests/kernel/mellanox_config.pm

```diff
@@ -47,27 +49,28 @@ sub run {
     assert_script_run("./" . $mft_version . "/install.sh");
     assert_script_run("mst start");
 
-    # List network devices
-    assert_script_run("lspci|egrep -i 'network|ethernet'");
-    if (script_run("lspci|grep -i mellanox") != 0) {
-        die "echo There is no Mellanox card here";
-    }
-    if (script_run("ls " . $device1) != 0) {
-        die "echo The directory $device1 doesn't exist";
-    }
-    if (script_run("ls " . $device2) != 0) {
-        die "echo The directory $device2 doesn't exist";
-    }
+    if (check_var('BACKEND', 'ipmi')) {
+        # List network devices
+        assert_script_run("lspci|egrep -i 'network|ethernet'");
+        if (script_run("lspci|grep -i mellanox") != 0) {
+            die "echo There is no Mellanox card here";
+        }
+        if (script_run("ls " . $device1) != 0) {
+            die "echo The directory $device1 doesn't exist";
+        }
+        if (script_run("ls " . $device2) != 0) {
+            die "echo The directory $device2 doesn't exist";
+        }
+        # Change Link protocol
+        script_run("echo Wanted Link protocol is $protocol");
+        show_links();
+        assert_script_run("mlxconfig -y -d $device1 set LINK_TYPE_P1=$protocol LINK_TYPE_P2=$protocol");
+        assert_script_run("mlxconfig -y -d $device2 set LINK_TYPE_P1=$protocol LINK_TYPE_P2=$protocol");
+        show_links();
 
-    # Change Link protocol
-    script_run("echo Wanted Link protocol is $protocol");
-    show_links();
-    assert_script_run("mlxconfig -y -d $device1 set LINK_TYPE_P1=$protocol LINK_TYPE_P2=$protocol");
-    assert_script_run("mlxconfig -y -d $device2 set LINK_TYPE_P1=$protocol LINK_TYPE_P2=$protocol");
-    show_links();
-
-    # Reboot system
-    power_action('reboot', textmode => 1, keepconsole => 1);
+        # Reboot system
+        power_action('reboot', textmode => 1, keepconsole => 1);
```

Comment:
let's reboot VM too?

Reply:
I tried but it didn't work, maybe we need a different command, or different console mode or something. Didn't want to investigate further, as it is not so important I think.